### PR TITLE
Add default error message when received empty error message for compilation failure

### LINKED ISSUE #708: Bug: Studio fails to navigate to error location in text mode if error does not contain error messgae
### Similar issues

- [X] I have searched and found no existing similar issues

### How are you using Studio?

Legend Studio

### Current and expected behavior

Use Grammar below in text mode. There is a compilation error without an error message. The error message contains source information. Because there is no error message Studio fails to navigate user to location of error message.

Although engine should return an error message, Studio should navigate users to where the error occurred.
Corresponding Engine Issue - https://github.com/finos/legend-engine/issues/455

Engine Response to Studio
```
{
  "errorType": "COMPILATION",
  "sourceInformation": {
    "endColumn": 156,
    "endLine": 5,
    "sourceId": "",
    "startColumn": 3,
    "startLine": 5
  },
  "stackTrace": [],
  "suppressed": []
}

```

### Steps to reproduce

Compile code below in text mode. 

### Model data

```text
Class model::Person
{
  firstName: String[1];
  lastName: String[1];
  isEqual(fullName: String[1], noCheck: Boolean[1]) {if($noCheck, |true, |if(($this.firstName + $this.lastName) == $fullNameL, |true, |false))}: Boolean[1];
}
```

### Environment

```text
Windows
```

### Possible solution and workaround

_No response_

### Contribution

- [ ] I would like to work on the fix for this issue

### PR DESCRIPTION
**Pull request by gayathrir11**
## Summary

Closes #708 

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

pic3

### CODE REVIEW

File: packages/legend-application/src/components/LambdaEditor.tsx [hunk 1]
```diff
@@ -325,6 +325,9 @@ const LambdaEditorInline = observer(
         editorModel.updateOptions({ tabSize: TAB_SIZE });
         const error = parserError ?? compilationError;
         if (error?.sourceInformation) {
+          if (error.message === '') {
```
Comment (MauricioUyaguari) [inline packages/legend-application/src/components/LambdaEditor.tsx:328]: Rethinking this I dont like us replacing the orginal value of the error. Could we may add a getter property to `EngineError` called nonEmptyMessage where we return either the orginal message if not empty and if empty `No error message provided by engine`.

File: .changeset/slow-kings-yell.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@finos/legend-studio": patch
+"@finos/legend-query": patch
+"@finos/legend-application": patch
+---
```
Comment (MauricioUyaguari) [inline .changeset/slow-kings-yell.md:5]: Handle empty error messages from Engine.

File: packages/legend-studio/src/components/editor/edit-panel/GrammarTextEditor.tsx [hunk 3]
```diff
@@ -199,7 +199,7 @@ export const GrammarTextEditor = observer(() => {
       if (error?.sourceInformation) {
         setErrorMarkers(
           editorModel,
-          error.message,
+          error.message === '' ? error.nonEmptyMessage : error.message,
```
Comment (akphi) [inline packages/legend-studio/src/components/editor/edit-panel/GrammarTextEditor.tsx:202]: I think we don't need this, let's just hard code the error here `(no error message)`

Comment (gayathrir11) [reply packages/legend-studio/src/components/editor/edit-panel/GrammarTextEditor.tsx:202]: will update it.

File: packages/legend-graph/src/graphManager/action/EngineError.ts [hunk 4]
```diff
@@ -34,6 +34,10 @@ export class EngineError extends ApplicationError {
   setSourceInformation(sourceInformation: SourceInformation | undefined): void {
     this.sourceInformation = sourceInformation;
   }
+
+  get nonEmptyMessage(): string {
```
Comment (akphi) [inline packages/legend-graph/src/graphManager/action/EngineError.ts:38]: don't think we need this, let's just hard code the error in various places with the message `(no error message)`

Comment (gayathrir11) [reply packages/legend-graph/src/graphManager/action/EngineError.ts:38]: will remove this

**Review by akphi — APPROVED** [review]
@MauricioUyaguari and I think the best way is to handle things like this at `ApplicationError` level, but we will take of it. Thanks a lot for doing this!

**Pull request merged**